internal/web_api.go

```diff
@@ -145,28 +150,59 @@ func (a *fileUploadAPI) UploadFile(filename string, file []byte) (string, error)
 }
 
 func (a *api) send(req *http.Request, respPayload interface{}) error {
-	resp, err := a.httpClient.Do(req)
-	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
-	bodyBytes, err := ioutil.ReadAll(resp.Body)
-	if resp.StatusCode != http.StatusOK {
-		apiErr := &api_errors.ErrAPI{}
-		if err := json.Unmarshal(bodyBytes, apiErr); err != nil {
-			return fmt.Errorf("couldn't unmarshal error response: %s (code: %d, raw body: %s)", err.Error(), resp.StatusCode, string(bodyBytes))
+	var attempts uint
+	var do func() error
+
+	do = func() error {
+		resp, err := a.httpClient.Do(req)
+		if err != nil {
+			return err
+		}
+		defer resp.Body.Close()
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if resp.StatusCode != http.StatusOK {
+			apiErr := &api_errors.ErrAPI{}
+			if err := json.Unmarshal(bodyBytes, apiErr); err != nil {
+				return fmt.Errorf("couldn't unmarshal error response: %s (code: %d, raw body: %s)", err.Error(), resp.StatusCode, string(bodyBytes))
+			}
+			if apiErr.Error() == "" {
+				return fmt.Errorf("couldn't unmarshal error response (code: %d, raw body: %s)", resp.StatusCode, string(bodyBytes))
+			}
+
+			if a.retryStrategy == nil || !a.retryStrategy(attempts, apiErr) {
+				return apiErr
+			}
+
+			token, err := a.getToken()
+			if err != nil {
+				return err
+			}
+
+			req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token.AccessToken))
```

Comment:
please use `Type` from `token`, we support also `Basic` :D

Reply:
good catch, i fixed also `UploadFile`